rhino-utils/upload_file_to_s3.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Check if the correct number of arguments are provided
+if [ "$#" -lt 3 ]; then
+    echo "Usage: $0 <upload_directory> <s3_bucket> <path_in_bucket> [aws_access_key_id] [aws_secret_access_key]"
+    exit 1
+fi
+
+# Assign the arguments to variables
+upload_directory=$1
+s3_bucket=$2
+path_in_bucket=$3
+
+# If AWS credentials are provided, use them. Otherwise, use the ones from the environment.
+if [ "$#" -ge 5 ]; then
+    aws_access_key_id=$4
+    aws_secret_access_key=$5
+else
+    aws_access_key_id=$AWS_ACCESS_KEY_ID
```

Comment:
What if these aren't defined or are empty? Should we check that and if they are empty/missing to notify the user (preferably with some guidance on what they need to do in order to set these)?